feat: new extension `vike-react-chakra` (#151)
vikejs/vike-react: packages/vike-react-chakra/README.md

> <!-- WARNING: keep links absolute in this file so they work on NPM too -->

Reviewer: To increase maintainability, let's further minimize the Readme. Let's remove the Vike logo.

[<img src="https://vike.dev/vike-readme.svg" align="right" height="90">](https://vike.dev)
[![npm version](https://img.shields.io/npm/v/vike-react-chakra)](https://www.npmjs.com/package/vike-react-chakra)

# `vike-react-chakra`

[Installation](#installation)  
[Basic usage](#basic-usage)  
[`+chakra` Setting](#chakra-setting)  
[See also](#see-also)  

<br/>

## Installation

> 1. npm install @chakra-ui/react @emotion/react vike-react-chakra

Reviewer: Backticks are missing. Let's put `vike-react-chakra` first in the install list.

2. Extend `+config.js`:
   ```js
   // pages/+config.js

   import vikeReact from 'vike-react/config'
   import vikeReactChakra from 'vike-react-chakra/config'

   export default {
     // ...
     extends: [vikeReact, vikeReactChakra]
   }
   ```

> [!NOTE]
> The `vike-react-chakra` extension requires [`vike-react`](https://vike.dev/vike-react).

<br/>

> ## Basic usage

Reviewer: Maybe we can remove the whole `Basic usage` section with something like the following. WDYT?

> That's it, you can now use Chakra components for any of your component.
>
> ```jsx
> import { HStack, Button } from '@chakra-ui/react'
> const Demo = () => {
>   return (
>     <HStack>
>       <Button>Click me</Button>
>       <Button>Click me</Button>
>     </HStack>
>   )
> }
> ```

Reviewer: So essentially removing the note and the heading.

```jsx
import { HStack, Button } from '@chakra-ui/react'

const Demo = () => {
  return (
    <HStack>
      <Button>Click me</Button>
      <Button>Click me</Button>
    </HStack>
  )
}
```

> [!NOTE]
> Chakra UI provides a cli tool to add snippets.  
> Snippets are pre-built components that you can use to build your UI faster.  
> Using the `@chakra-ui/cli` you can add snippets to your project.

```
npx @chakra-ui/cli snippet add
```

<br/>

> ## `+chakra` Setting

Reviewer: ```suggestion
## `+chakra` Setting

```

> `vike-react-chakra` provides a custom setting, `+chakra`, for configuring custom theme system, custom locale to adds `LocaleProvider` and can be used to disable Chakra UI for specific pages.

Reviewer: ```suggestion
`vike-react-chakra` provides a configuration `+chakra` for setting a custom theme system and custom locale.
```

Let's mention `can be used to disable Chakra UI for specific pages` later.

```js

> // pages/+chakra.js

Reviewer: ```suggestion
// pages/+chakra.js

```

> import React, { ReactNode } from 'react'

Reviewer: ```suggestion
import React, { type ReactNode } from 'react'
```

import { createSystem, defaultConfig, defineConfig } from '@chakra-ui/react'

const customConfig = defineConfig({
  globalCss: {
    "html, body": {
      margin: 0,
      padding: 0,
    },
  },
})

const system = createSystem(defaultConfig, customConfig)

export const chakra = {
  system,

>   locale: "ar-BH"

Reviewer: Let's showcase one that is more widely known e.g. `fr-FR`.

}
```

> To disable Chakra UI for specific page:

Reviewer: ```suggestion
You can remove Chakra UI [for some of your pages](https://vike.dev/config#inheritance):
```

```js

> // pages/about/+chakra.js

Reviewer: ```suggestion
// pages/about/+chakra.js

```

export const chakra = null
```

<br/>

> ## See also

Reviewer: Good idea but I think it's better we remove it to keep things maintainable; we will have a *lot* of `vike-react-*` packages.
